File: 3__protected.cpp


#include <bits/stdc++.h>
using namespace std;

class Person
{
protected:
    string name; // protected variable

public:
    void setName(string n)
    {
        name = n;
    }
    string getName()
    {
        return name;
    }
};

int main()
{

    Person p1;
    p1.setName("meraz");
    cout << p1.getName();

   
    return 0;
}

File: 3__protected.py
class Person:
    def __init__(self,name):
        self._name = name
        
    def get_name(self):
        return self._name 
    
    
person = Person("Meraz")
print(person.get_name() if person.get_name() else "This name is protect")


print(person._name) #here also wil be a output
